server/sonar-server/src/main/java/org/sonar/server/permission/PermissionFinder.java

```diff
@@ -19,141 +19,75 @@
-import org.sonar.core.permission.GlobalPermissions;
 import org.sonar.db.DbClient;
 import org.sonar.db.DbSession;
-import org.sonar.db.component.ResourceDao;
-import org.sonar.db.component.ResourceDto;
-import org.sonar.db.component.ResourceQuery;
-import org.sonar.db.permission.GroupWithPermissionDto;
 import org.sonar.db.permission.PermissionDao;
 import org.sonar.db.permission.PermissionQuery;
-import org.sonar.db.permission.UserWithPermissionDto;
-import org.sonar.server.exceptions.NotFoundException;
+import org.sonar.db.user.GroupDao;
+import org.sonar.db.user.GroupDto;
+import org.sonar.db.user.GroupRoleDto;
+import org.sonar.db.user.UserDao;
+import org.sonar.db.user.UserDto;
+import org.sonar.db.user.UserPermissionDto;
 
-import static com.google.common.collect.Lists.newArrayList;
-import static org.apache.commons.lang.StringUtils.containsIgnoreCase;
-import static org.sonar.api.utils.Paging.forPageIndex;
+import static java.util.Collections.emptyList;
 
 @ServerSide
 public class PermissionFinder {
 
   private final PermissionDao permissionDao;
-  private final ResourceDao resourceDao;
+  private final UserDao userDao;
+  private final GroupDao groupDao;
 
   public PermissionFinder(DbClient dbClient) {
-    this.resourceDao = dbClient.resourceDao();
     this.permissionDao = dbClient.permissionDao();
+    this.userDao = dbClient.userDao();
+    this.groupDao = dbClient.groupDao();
   }
 
-  public List<UserWithPermissionDto> findUsersWithPermission(DbSession dbSession, PermissionQuery query) {
-    Long componentId = componentId(query.component());
-    int limit = query.pageSize();
-    return permissionDao.selectUsers(dbSession, query, componentId, offset(query), limit);
-  }
+  public List<UserDto> findUsers(DbSession dbSession, PermissionQuery.Builder dbQuery) {
+    List<String> orderedLogins = permissionDao.selectLoginsByPermissionQuery(dbSession, dbQuery.build());
 
-  /**
-   * Paging for groups search is done in Java in order to correctly handle the 'Anyone' group
-   */
-  public List<GroupWithPermissionDto> findGroupsWithPermission(DbSession dbSession, PermissionQuery query) {
-    Long componentId = componentId(query.component());
-    return toGroupQueryResult(permissionDao.selectGroups(dbSession, query, componentId), query);
+    return Ordering.explicit(orderedLogins).onResultOf(UserDto::getLogin).immutableSortedCopy(userDao.selectByLogins(dbSession, orderedLogins));
   }
 
-  @Nullable
-  private Long componentId(@Nullable String componentKey) {
-    if (componentKey == null) {
-      return null;
-    } else {
-      ResourceDto resourceDto = resourceDao.selectResource(ResourceQuery.create().setKey(componentKey));
-      if (resourceDto == null) {
-        throw new NotFoundException(String.format("Project '%s' does not exist", componentKey));
-      }
-      return resourceDto.getId();
+  public List<UserPermissionDto> findUserPermissions(DbSession dbSession, PermissionQuery.Builder dbQuery, List<UserDto> users) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Document this public method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AUndocumentedApi)